src/components/container/GlucoseChart/GlucoseChart.tsx

```diff
@@ -0,0 +1,224 @@
+            add(selectedMeal.timestamp, { minutes: 60 }).valueOf(),
+            add(selectedMeal.timestamp, { minutes: 90 }).valueOf()
+        ];
+        chartTickFormatter = (value: number) => format(new Date(value), 'h:mmaaa');
+    }
+
+    const customDot = (props: { cx: number, cy?: number, payload: { timestamp: Date, meal?: boolean } }) => {
+        const { cx, cy, payload } = props;
+        if (!cy || !payload.meal) return <></>;
+
+        let mealIndex = meals.findIndex(meal => meal.timestamp === payload.timestamp);
+        if (mealIndex < 0) return <></>;
+
+        return <svg>
+            <circle cx={cx} cy={cy} r={6} fill={getColorFromAssortment(mealIndex)} />
+        </svg>;
+    };
+
+    const customDotLabel = (props: any) => {
+        const { x, y, index } = props;
+
+        let entry = chartData[index];
+        if (!entry.meal) return <></>;
+
+        let mealIndex = meals.findIndex(meal => meal.timestamp === entry.timestamp);
+        if (mealIndex < 0) return <></>;
+
+        return <text x={x} y={y} dy={3} fill="#fff" fontSize={8} textAnchor="middle">{mealIndex + 1}</text>;
+    };
+
+    let filteredSteps = steps?.filter(reading => reading.value > 0) ?? [];
+    let maxSteps = filteredSteps.length > 0 ? Math.max(...filteredSteps.map(r => r.value)) : 0;
+    let stepsScale = maxSteps > 0 ? 240 / maxSteps : 1;
+    let overlaySteps = filteredSteps.map(r => ({ ...r, value: r.value * stepsScale }));
+
+    return <div className="mdhui-glucose-chart" ref={props.innerRef}>
+        <div className="mdhui-glucose-chart-chart" style={{ display: !loading && glucose && glucose.length > 0 ? 'block' : 'none' }}>
+            <TimeSeriesChart
+                intervalType="Day"
+                intervalStart={selectedDate}
+                data={chartData as any}
+                series={[{ dataKey: 'value', color: '#999' }]}
+                chartHasData={!!glucose && glucose.length > 0}
+                chartType="Line"
+                options={{
+                    lineOptions: {
+                        dot: customDot,
+                        label: customDotLabel,
+                        strokeWidth: 1.5,
+                        animationDuration: 500
+                    },
+                    containerOptions: {
+                        height: 166
+                    },
+                    xAxisOptions: {
+                        domain: chartDomain,
+                        ticks: chartTicks,
+                        tickFormatter: chartTickFormatter
+                    },
+                    yAxisOptions: {
+                        width: 24,
+                        domain: [0, 240],
+                        ticks: [40, 80, 120, 160, 200, 240]
+                    }
+                }}
+            >
+                {avgGlucose &&
+                    <ReferenceLine
+                        y={avgGlucose}
+                        stroke="var(--mdhui-color-primary)"
```

Comment:
To make this easier to digest on first sight, consider making the icons for blood glucose (range and avg) different colors, and coloring the reference line for avg glucose to match the icon color. 
Atm, the reference line for avg glucose is colored closer to sleep icon.

Reply:
I think we'll want to keep the two blood icons the same color since they both refer to the same data.

The average line is currently using the primary color (which is configurable), but I will add a configuration option for that in case someone wants to override the primary color with something else.

Reply:
I've added an optional property on the `GlucoseChart` for the average glucose reference line color.